lib/client/doc.js

```diff
@@ -919,6 +919,17 @@ Doc.prototype.resume = function() {
   this.flush();
 };
 
+// Create a snapshot that can be serialized, deserialized, and passed into `Doc.ingestSnapshot`.
+//
+// Note that types.defaultType is used inside ingestSnapshot when snapshot.type is omitted.
+// If you need to use a different type, populate `snapshot.type` with the OT type you are using.
+Doc.prototype.createSnapshot = function() {
+  return {
+    v: this.version,
+    data: this.data,
```

Comment:
We should probably clone `data`, because if an op comes in, it'll actually mutate the snapshot's data (since they share a reference), which should never change.

Can we also add a test case for this, so we don't accidentally do this in future?

Reply:
Added a failing test.